src/ctapipe/image/toymodel.py

```diff
@@ -440,7 +440,7 @@ def linear_pdf(self, var, var_shift, linear_slope=0):
             raise ValueError("The var array is too short.")
 
         if np.isnan(var).any():
-            raise ValueError("array contains nans.")
+            raise Warning("array contains nans.")
```

Comment:
This is still an exception, since you `raise` it. To warn, you need `warnings.warn` and you can pass the class. You shouldn't use the base `Warning`.

However, I disagree and would like to understand where in normal operations, there can be a nan here. I thought we excluded that?